interpreter/cling/lib/Utils/ASTImportSource.cpp

```diff
@@ -0,0 +1,189 @@
+        importer.ImportContext(declContextToImport);
+
+      if (importedDeclContext) {
+        /* And also put the declaration context I found from the parent Interpreter
+         * in the map of the child Interpreter to have it for the future. */
+        m_DeclContexts_map[importedDeclContext] = declContextToImport;
+
+        /* Not used for now.
+         * m_DeclContextsNames_map[childDeclName.getAsString()] =
+         * std::make_pair(importedDeclContext,declContextToImport); */
+        importedDeclContext->setHasExternalVisibleStorage(true);
+
+        std::vector < NamedDecl * > declVector;
+        declVector.push_back((NamedDecl *) importedDeclContext);
+        llvm::ArrayRef < NamedDecl * > FoundDecls(declVector);
+        SetExternalVisibleDeclsForName(childCurrentDeclContext,
+                                       ((NamedDecl *) importedDeclContext)->getDeclName(),
+                                       FoundDecls);
+
+        /* And also put the Decl I found from the parent Interpreter
+         * in the map of the child Interpreter to have it for the future. */
+        m_DeclName_map[childDeclName.getAsString()] = parentDeclName;
+      }
+    }
+
+    bool ASTImportSource::Import(DeclContext::lookup_result lookup_result,
+                                 ASTContext &from_ASTContext,
+                                 ASTContext &to_ASTContext,
+                                 const DeclContext *childCurrentDeclContext,
+                                 DeclarationName &childDeclName,
+                                 DeclarationName &parentDeclName) {
+
+      /* Check if we found this Name in the parent interpreter */
+      if (lookup_result.empty())
+        return false;
+
+      /* Prepare to import the Decl(Context)  we found in the
+       * child interpreter */
+      const FileSystemOptions systemOptions;
+      FileManager fm(systemOptions, nullptr);
+
+      /*****************************ASTImporter**********************************/
+      ASTImporter importer(to_ASTContext, fm, from_ASTContext, fm,
+        /*MinimalImport : ON*/ true);
+      /**************************************************************************/
+
+      /* If this Name we are looking for is for example a Namespace,
+       * then it is a Decl Context. */
+      if ((*lookup_result.data())->getKind() == Decl::Namespace) {
+        DeclContext *declContextToImport =
+          llvm::cast<DeclContext>(*lookup_result.data());
+        ImportDeclContext(declContextToImport, importer, childDeclName,
+                          parentDeclName, childCurrentDeclContext);
+      } else { /* If this name is just a Decl. */
+        /* Check if we have more than one results, this means that it
+         * may be an overloaded function. */
+        if (lookup_result.size() > 1) {
+          DeclContext::lookup_iterator I;
+          DeclContext::lookup_iterator E;
+          for (I = lookup_result.begin(), E = lookup_result.end(); I != E; ++I) {
```

Comment:
It's nicer to declare at first use, i.e. `for (DeclContext::lookup_iterator I = lookup_result.begin(), E = lookup_result.end()`...